src/ui/components/Timeline/index.js

```diff
@@ -298,10 +293,12 @@ export class Timeline extends Component {
       nextGraphicsTime = nextPaintOrMouseEvent(currentTime)?.time || this.zoomEndTime;
       nextGraphicsPromise = getGraphicsAtTime(nextGraphicsTime);
     };
-
     prepareNextGraphics();
 
-    while (this.props.playback) {
+    // This is a `do ... while` instead of `while` because startPlayback is synchronous.
+    // It's possible that the playback in state hasn't been updated yet by the time
+    // this is first called. So we just force it to run the first time.
+    do {
```

Comment:
i can see why you wanted to clean this up and move it to redux :)

Reply:
Will do!